contracts/SelfClaimERC20BudgetApproval.sol

```diff
@@ -0,0 +1,345 @@
+// SPDX-License-Identifier: GPL-3.0
+
+pragma solidity 0.8.7;
+
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@openzeppelin/contracts/utils/Counters.sol";
+
+import "./lib/BytesLib.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import "./interface/IBudgetApprovalExecutee.sol";
+import "@chainlink/contracts/src/v0.8/Denominations.sol";
+import "hardhat/console.sol";
+
+
+contract SelfClaimERC20BudgetApproval is Initializable {
+  using BytesLib for bytes;
+  using Counters for Counters.Counter;
+
+  string public constant name = "SelfClaim ERC20 Budget Approval";
+
+  // !!Caution Approved and Cancelled should not be used in this budgetApproval
+  enum Status {
+    Pending,
+    Approved,
+    Completed,
+    Cancelled
+  }
+
+  struct Transaction {
+    uint256 id;
+    bytes[] data;
+    Status status;
+    uint32 deadline;
+    bool isExist;
+  }
+
+  event CreateTransaction(
+    uint256 indexed id,
+    bytes[] data,
+    uint256 deadline,
+    Status status,
+    string comment,
+    address creator
+  );
+
+  event ExecuteTransaction(
+    uint256 indexed id,
+    bytes[] data,
+    address _executor
+  );
+
+  event AllowAddress(address target);
+  event AllowToken(address token);
+  event AllowAmount(uint256 amount);
+  event ExecuteSelfClaimERC20Transaction(
+    uint256 indexed id,
+    address indexed executor,
+    address indexed toAddress,
+    address token,
+    uint256 amount
+  );
+  
+  Counters.Counter private _transactionIds;
+
+  mapping(uint256 => Transaction) public transactions;
+
+  address private _executee;
+  string private _text;
+  string private _transactionType;
+
+  bool private _allowUnlimitedUsageCount;
+  uint256 private _usageCount;
+
+  uint256 private _startTime;
+  uint256 private _endTime;
+
+  // Any receiver
+  bool public allowAllAddresses;
+  mapping(address => bool) public addressesMapping;
+  // only support single ERC20 token
+  address public token;
+  bool public allowAnyAmount;
+  uint256 public totalAmount;
+  uint256 public fixClaimAmount;
+
+  mapping(address => bool) private claimedAddrMapping;
+
+  struct InitializeParams {
```

Comment:
I edited the CommonBudgetApproval to use _isExecutor() insteads of get the variable from storage
Check this example out, and you may can inherit from CommonBudgetApproval again and  remove unnecessary code
https://github.com/adam-vault/adam-contract-core/blob/main/contracts/DepositRewardBudgetApproval.sol